packages/openactive-integration-tests/test/helpers/flow-stages/opportunity-feed-update.js

```diff
@@ -0,0 +1,171 @@
+const { getRelevantOffers } = require('@openactive/test-interface-criteria');
+const chakram = require('chakram');
+const sharedValidationTests = require('../../shared-behaviours/validation');
+const { isResponse20x } = require('../chakram-response-utils');
+
+/**
+ * @typedef {import('chakram').ChakramResponse} ChakramResponse
+ * @typedef {import('../../types/OpportunityCriteria').OpportunityCriteria} OpportunityCriteria
+ * @typedef {import('../logger').BaseLoggerType} BaseLoggerType
+ * @typedef {import('../request-helper').RequestHelperType} RequestHelperType
+ */
+
+const { HARVEST_START_TIME } = global;
+
+/**
+ * @typedef {{
+ *   position: number;
+ *   orderedItem: {
+ *     '@type': string,
+ *     '@id': string,
+ *     [k: string]: any,
+ *   };
+ *   acceptedOffer: {
+ *     '@id': string,
+ *     [k: string]: any,
+ *   };
+ *   'test:primary': boolean;
+ *   'test:control': boolean;
+ * }} OrderItem
+ */
+
+/**
+ * @param {unknown} opportunity
+ * @param {string} opportunityCriteria
+ */
+function getRandomRelevantOffer(opportunity, opportunityCriteria) {
+  const relevantOffers = getRelevantOffers(opportunityCriteria, opportunity, { harvestStartTime: HARVEST_START_TIME });
+  if (relevantOffers.length === 0) { return null; }
+
+  return relevantOffers[Math.floor(Math.random() * relevantOffers.length)];
+}
+
+/**
+ * Wait for an opportunity to appear in the Booking System's Opportunity Feed.
+ */
+const OpportunityFeedUpdateFlowStage = {
```

Comment:
I haven't set-up a `create(..)` function for this flow stage yet as this stage is not needed for the accept-proposal-book test.

However, the `run()` function has been coded as it is used by `fetch-opportunities`